### DISCUSSION TITLE
Implement Testing

### DISCUSSION THREAD

**Discussion by jtfairbank** [distributeaid/distributeaid.org#588] [Ideas]
I think it's a good idea to implement testing given the fact that I broke production a few times recently, and that we have a bunch of new contributors.  Does anyone have any thoughts on how to best approach this for a Gatsby / React site?  I'd like to keep it simple enough that our new contributors can pick it up fairly easily.

@coderbyheart I know you do some really cool testing-in-prod stuff!  Would that be relevant here?

**Comment by raae** [comment]
I had success with [the docs on testing](https://www.gatsbyjs.com/docs/how-to/testing/) last time I set up testing for a Gatsby site.

**Comment by coderbyheart** [comment]
Because this is a content heavy website and there is little logic to test (which also has some complicated dependencies to set up), I'd also recommend to use an end-to-end testing framework, as @raae already did 

[Playwright](https://playwright.dev/)  is very easy to get started, very fast, designed for SPA (which Gatsby is not, but our website has content that is displayed using Client-Side JavaScript) and I have [working examples](https://github.com/NordicSemiconductor/asset-tracker-cloud-app-aws-js/blob/56c584133eaec44f66471e3635dd88468402da7d/.github/workflows/test-and-release.yaml) for GitHub actions.

**Reply by coderbyheart** [reply]
Since I had previous work, I've added a demonstration in #589
